### ISSUE TITLE
access web console on different port instead of 8443

### ISSUE THREAD

**Issue by priyanka5** [openshift/origin#3039]
Hi , is there a way I can access web console on different port instead of 8443, because my firewall block this port :(

**Comment by liggitt** [comment]
If you cannot access 8443, you would also need to change the port for the API. You can change the port the API server and the UI are started on by setting `--listen=https://0.0.0.0:1234` when starting OpenShift

**Comment by Yashu5** [comment]
Hi , Thanks a lot for response , I am starting origin server as docker container itself as explained in docs, you mean port can be changed if I run container using : 

docker run -d --name "openshift-origin" --net=host --privileged \
    -v /var/run/docker.sock:/var/run/docker.sock \
    openshift/origin start --listen=https://0.0.0.0:1234


please let me know if I am wrong.

Thanks, yash

**Comment by liggitt** [comment]
That will change the port openshift listens on... I'm not sure if that will expose the port through the container. You also may need to specify the external hostname using `--public-master=https://external-host:1234` if the web browser accesses the container using a different IP/host than is detected internally

**Comment by Yashu5** [comment]
Hi, apologies for replying late and on closed thread but whenever I start my openshift server to listen to port 8080 using : ./openshift start --listen=localhost:8080 , it does not seem to be working. In logs it shows -
Unable to start etcd: listen tcp localhost:7001: bind: cannot assign requested address

any thoughts on this would be appreciated??

**Comment by liggitt** [comment]
try the local IP of your system, e.g. --listen=https://192.168.1.101:8080

**Comment by Yashu5** [comment]
i am running it in AWS ec2 instance and have tried using it with private and public ip addresses both, but no luck :( it just stucks at-

 Unable to start etcd: listen tcp 172.89.09.6:7001: bind: address already in use

**Comment by Yashu5** [comment]
it seems to be working now :) i wonder why it was not resolving localhost to my machines ip

**Comment by rushins** [comment]
i dont have firewall issue however i cannot access the web console https://ip:8443/ but from linux side the "oc" commands working.

does anyone have any idea what to do no this.

**Comment by cmoulliard** [comment]
Is there a doc/note explaining for recent versions of OpenShift what should be done during installation to configure it to use a different port number ? Is there an option part of the OpenShift Playbook to configure the port to be used (e.g 8443 -> 80, ....) ?

**Comment by geoand** [comment]
@cmoulliard [This](https://docs.openshift.com/container-platform/3.9/install_config/install/advanced_install.html#configuring-host-port) part of the doc is probably applicable.

However it doesn't tell us what to do to change the port of an already running cluster :(

**Comment by cmoulliard** [comment]
> cluster

And the problem could be that this ansible doc

**Comment by mfojtik** [comment]
I guess editing the master config manually and replacing the port should do the trick, maybe @liggitt will be more skeptical ;-)

You can also change that in the ansible hosts file and rerun the playbook? /cc @sdodson

**Comment by geoand** [comment]
Changing the port manually doesn't work because then the certificate doesn't match.

Running some playbook would be nice, but which playbook would that be?

**Comment by sdodson** [comment]
> You can also change that in the ansible hosts file and rerun the playbook? /cc @sdodson

Yes, re-running playbooks/deploy_cluster.yml on >= 3.9 or playbooks/byo/config.yml on <= 3.7 will reconfigure this.
https://docs.openshift.com/container-platform/3.9/install_config/install/advanced_install.html#configuring-host-port

I don't think the original question was based on an openshift-ansible install however.
